sites/all/modules/custom/lesson_entity/lesson_entity.module

```diff
@@ -0,0 +1,330 @@
+ */
+function lesson_entity_basic_list_entities() {
+  $content = [];
+  $entities = lesson_entity_basic_load_multiple();
+  $header = [
+    ['data' => 'ID', 'field' => 'id'],
+    ['data' => 'UID', 'field' => 'uid'],
+    ['data' => 'Title', 'field' => 'title'],
+    ['data' => 'Body', 'field' => 'body'],
+    ['data' => 'type', 'field' => 'type'],
+  ];
+
+  if (!empty($entities)) {
+    foreach ($entities as $entity) {
+      $rows[] = [
+        'data' => [
+          'id' => $entity->id,
+          'uid' => $entity->uid,
+          'title' => $entity->title,
+          'body' => $entity->body,
+          'type' => $entity->type,
+        ],
+      ];
+    }
+    $content['entity_table'] = [
+      '#theme' => 'table',
+      '#rows' => $rows,
+      '#header' => $header,
+    ];
+  }
+  else {
+    $content[] = [
+      '#type' => 'item',
+      '#markup' => t('No lesson_entity_basic entities currently exist.'),
+    ];
+  }
+
+  return $content;
+}
+
+/**
+ * Menu callback to display an entity.
+ */
+function lesson_entity_basic_show($entity) {
+  $header = [
+    t('Entity Property'),
+    t('Value'),
+  ];
+  $rows = [];
+  foreach ($entity as $property => $value) {
+    if ($property !== 'rdf_mapping') {
+      $rows[] = [
+        'data' => [$property, $value],
+      ];
+    }
+  }
+
+  return theme('table', $variables = [
+    'header' => $header,
+    'rows' => $rows,
+    'caption' => t('Entity'),
+  ]);
+}
+
+/**
+ * Provides a wrapper on the edit form to add a new entity.
+ */
+function lesson_entity_basic_add() {
+  $entity = entity_create('lesson_entity_basic', ['title' => '', 'body' => '', 'type' => '']);
+  return drupal_get_form('lesson_entity_basic_form', $entity);
+}
+
+/**
+ * Form function to create an entity_example_basic entity.
+ */
+function lesson_entity_basic_form($form, &$form_state, $entity) {
+  $form['item_title'] = [
+    '#type' => 'textfield',
+    '#title' => t('Item Description'),
+    '#required' => TRUE,
+    '#default_value' => $entity->title,
+  ];
+
+  $form['item_body'] = [
+    '#type' => 'textfield',
+    '#title' => t('Item Body'),
+    '#required' => TRUE,
+    '#default_value' =>  $entity->body,
+  ];
+
+  $form['item_type'] = [
+    '#type' => 'select',
+    '#title' => t('Type'),
+    '#options' => [
+      1 => 1,
+      2 => 2,
+    ],
+    '#default_value' =>  $entity->type,
+  ];
+
+  $form['basic_entity'] = [
+    '#type' => 'value',
+    '#value' => $entity,
+  ];
+
+  $form['submit'] = [
+    '#type' => 'submit',
+    '#value' => t('Save'),
+    '#weight' => 100,
+  ];
+
+  $form['delete'] = [
+    '#type' => 'submit',
+    '#value' => t('Delete'),
+    '#submit' => ['lesson_entity_basic_edit_delete'],
+    '#weight' => 200,
+  ];
+
+  return $form;
+}
+
+
+/**
+ * Form submit handler: Submits information from entity creation form.
+ */
+function lesson_entity_basic_form_submit($form, &$form_state) {
+  $entity_fields = $form_state['values']['basic_entity'];

```

Comment:
You can use the way to simplify:
$values = $form_state['values'];
$entity['title'] = $values['item_title'];
..